rita/src/rita_common/traffic_watcher/mod.rs

```diff
@@ -126,16 +126,52 @@ pub fn watch<T: Read + Write>(mut babel: Babel<T>, neighbors: &Vec<Neighbor>) ->
     destinations.insert(SETTING.get_network().own_ip, Int256::from(0));
 
     trace!("Getting input counters");
-    let input_counters = KI.read_counters(&FilterTarget::Input)?;
-    info!("Got output counters: {:?}", input_counters);
-
-    trace!("Getting destination counters");
-    let output_counters = KI.read_counters(&FilterTarget::Output)?;
-    info!("Got destination counters: {:?}", output_counters);
+    let input_counters = match KI.read_counters(&FilterTarget::Input) {
+        Ok(res) => res,
+        Err(e) => {
+            warn!(
+                "Error getting input counters {:?} traffic has gone unaccounted!",
+                e
+            );
+            return Err(e);
+        }
+    };
+    trace!("Got input counters: {:?}", input_counters);
+
+    trace!("Getting ouput counters");
+    let output_counters = match KI.read_counters(&FilterTarget::Output) {
+        Ok(res) => res,
+        Err(e) => {
+            warn!(
+                "Error getting output counters {:?} traffic has gone unaccounted!",
+                e
+            );
+            return Err(e);
+        }
+    };
+    trace!("Got output counters: {:?}", output_counters);
 
     trace!("Getting fwd counters");
-    let fwd_input_counters = KI.read_counters(&FilterTarget::ForwardInput)?;
-    let fwd_output_counters = KI.read_counters(&FilterTarget::ForwardOutput)?;
+    let fwd_input_counters = match KI.read_counters(&FilterTarget::ForwardInput) {
+        Ok(res) => res,
+        Err(e) => {
+            warn!(
+                "Error getting input counters {:?} traffic has gone unaccounted!",
+                e
+            );
+            return Err(e);
+        }
+    };
+    let fwd_output_counters = match KI.read_counters(&FilterTarget::ForwardOutput) {
```

Comment:
Wouldn't `KI.read_counters(...).map_err(|e| { warn!("foo"); e })?` be shorter?

Reply:
well I didn't know about map err that's nice. But In general I don't put a very high value on brevity in code. I would rather make it twice as long to increase readability especially since the compiler is going to boil it all down to the same thing anyways.